cns/types/infiniband/errorcodes.go

```diff
@@ -0,0 +1,13 @@
+package infiniband
+
+type ErrorCode int
+
+const (
+	Success                  ErrorCode = 0
+	PodNotFound              ErrorCode = 1
+	DeviceUnavailable        ErrorCode = 2
+	DeviceNotFound           ErrorCode = 3
+	AnnotationNotFound       ErrorCode = 4
+	PodAlreadyAllocated      ErrorCode = 5
```

Comment:
how would we determine this? should this be idempotent, or if we want to return that pod-request is already in-progress let's return that

Reply:
I would say our response would be `200` so yeah idempotent

Reply:
Would you like me to name this "ConflictingPodInfo" or something like that? (or "ConflictingDeviceInfo"?)

Reply:
try and omit error codes entirely (because you already have the HTTP ones). `PodAlreadyAllocated` could be covered by returning HTTP/409 with an error message, `DeviceNotFound` is simply a 404, etc.
aside, PodAlreadyAllocated is the wrong naming because the _Pod_ is not allocated, the IBDev is.

Reply:
not conflicting, maybe something that says pod is being allocated or in-progress, yeah whether devices given are same or different - if the given pod-id is in-progress we should return that & not take a new request for it

Reply:
> Yeah I see your point about trying to get rid of error codes entirely However, how would we distinguish between `PodNotFound` vs `DeviceNotFound`?
> 
> > the Pod is not allocated, the IBDev is
> 
> The case I'm trying to cover here is more like "Pod A has already been assigned devices B and C, but you're now trying to assign it devices D and E"

I think this smells because the api structure isn't mapping to the objects conceptually right, left a comment on the swagger with more thoughts.